solana/pyth2wormhole/program/src/attest.rs

```diff
@@ -268,47 +268,112 @@ pub fn attest(ctx: &ExecutionContext, accs: &mut Attest, data: AttestData) -> So
     })?;
 
     // Adjust message account size if necessary
-    if !accs.wh_message.is_initialized() && accs.wh_message.payload.len() != payload.len() {
-        // Learn how much we're adjusting (Note: assuming surrounding
-        // Wormhole message type is constant-size).
-        // positive -> grow,
-        // negative -> shrink,
-        let diff = payload.len() as i64 - accs.wh_message.payload.len() as i64;
-        accs.wh_message.0 .0 .0.realloc(
-            (accs.wh_message.payload.len() as i64 + diff) as usize,
-            false,
-        )?;
-    }
-
-    // Send payload
-    let post_message_data = (
-        bridge::instruction::Instruction::PostMessage,
-        PostMessageData {
-            nonce: 0, // Superseded by the sequence number
-            payload,
-            consistency_level: data.consistency_level,
-        },
-    );
-
-    let ix = Instruction::new_with_bytes(
-        accs.config.wh_prog,
-        post_message_data.try_to_vec()?.as_slice(),
-        vec![
-            AccountMeta::new(*accs.wh_bridge.key, false),
-            AccountMeta::new(*((accs.wh_message.0).0).0.key, true),
-            AccountMeta::new_readonly(*accs.wh_emitter.key, true),
-            AccountMeta::new(*accs.wh_sequence.key, false),
-            AccountMeta::new(*accs.payer.key, true),
-            AccountMeta::new(*accs.wh_fee_collector.key, false),
-            AccountMeta::new_readonly(*accs.clock.info().key, false),
-            AccountMeta::new_readonly(solana_program::sysvar::rent::ID, false),
-            AccountMeta::new_readonly(solana_program::system_program::id(), false),
-        ],
-    );
-
-    trace!("Before cross-call");
+    let ix = if accs.wh_message.is_initialized() {
```

Comment:
I couldn't understand that why we are using reliable message post when account is not initialized. I think unreliable_post_message also creates an account if needed:
https://github.com/wormhole-foundation/wormhole/blob/dev.v2/solana/bridge/program/src/api/post_message.rs#L160

Reply:
Nice, this simplifies the critical path. Thanks!

Reply:
Done, this looks so much nicer now!